discovery/gossiper.go

```diff
@@ -435,13 +435,18 @@ type channelUpdateID struct {
 }
 
 // msgWithSenders is a wrapper struct around a message, and the set of peers
-// that oreignally sent ius this message. Using this struct, we can ensure that
+// that originally sent is this message. Using this struct, we can ensure that
 // we don't re-send a message to the peer that sent it to us in the first
 // place.
 type msgWithSenders struct {
 	// msg is the wire message itself.
 	msg lnwire.Message
 
+	// timestamp is the timestamp of the message. We store it
+	// here such that we don't have the cast the lnwire.Message
+	// to read its timestamp.
+	timestamp uint32
```

Comment:
Don't see the use of this. When you're actually de-duping the announcements, you need to type switch to grab the underlying type. At that point, the timestamp is already exposed.